src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -1027,75 +853,80 @@ public void resetCurrentStructure()
         currentStructure = null;
     }
 
+    //Unable to merge both loops since they rely on the end of the other.
+    @SuppressWarnings("squid:S3047")
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Suppressing the 'squid:S3047' warning is not allowed [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)